scripts/mafia.js

```diff
@@ -106,55 +110,69 @@ function Mafia(mafiachan) {
     }
     function htmlVoteTheme(mess) {
         return("<a href=\"po:send//votetheme " + mess + "\">" + mess + "</a>");
     }
-    function getBotName(name) {
-        if (name) {
-            return name;
-        }
-        var botName = "Game";
-        if (("theme" in mafia) && (mafia.theme !== undefined)) {
-			if ("botName" in mafia.theme) {
-				botName = mafia.theme.botName;
-			}
-		}
-        return botName;
-    }
     /*Sends a Game bot message, if no bot name is defined, it adds "±Game"
     * Note: use "srcname" instead of "src" as src holds the player's ID*/
-    function gamemsg(src, mess, botName, channel) {
-        if (mess === undefined || mess.length === 0) {
+    function gamemsg(src, mess, botName, channel, html) {
+        if (mess === undefined || mess === "") {
             return false;
         }
         var id = sys.id(src);
         if (id === undefined) {
             return false;
         }
-        if (mess.indexOf("***") === -1 && mess.indexOf("±") === -1 && mess.lastIndexOf(":") !== (parseInt(mess.length, 10) - 1) && mess.substring(0, Config.Mafia.max_name_length + 1).indexOf(":") === -1) {
-            mess = "±" + getBotName(botName) + ": " + mess;
+        if (mess === null) {
+            mess = "";
+        }
+        var pos = mess.indexOf(': ');
+        if (!botName && pos !== -1 && (mess.indexOf("±") !== -1 || mess.substring(0, Config.Mafia.max_name_length + 1).indexOf(":") !== -1)) {
+            var name = mess.substring(0, pos),
+                pid = sys.id(name),
+                message = html_escape(mess.substr(pos + 2)),
+                color = mafia.bot.color;
+            if (pid !== undefined) {
+                color = script.getColor(pid);
+            }
+            mess = "<font color='" + color + "'><timestamp/> <b>" + name + ":</b></font> " + message;
+        } else if (mess.indexOf("***") === -1) {
+            mess = "<font color='" + mafia.bot.color + "'><timestamp/> <b>" + (botName ? botName : "±" + mafia.bot.name) + ":</b></font> " + (html ? mess : html_escape(mess));
         }
         if (channel === undefined) {
             channel = mafiachan;
         }
-        sys.sendMessage(id, mess, channel);
+        sys.sendHtmlMessage(id, mess, channel);
         return true;
     }
-    function gamemsgAll(mess, botName, channel) {
-        if (mess === undefined || mess.length === 0) {
+    function gamemsgAll(mess, botName, channel, html) {
+        if (mess === undefined || mess === "") {
             return false;
         }
-        if (mess.indexOf("***") === -1 && mess.indexOf("±") === -1 && mess.lastIndexOf(":") !== (parseInt(mess.length, 10) - 1) && mess.substring(0, Config.Mafia.max_name_length + 1).indexOf(":") === -1) {
-            mess = "±" + getBotName(botName) + ": " + mess;
+        if (mess === null) {
+            mess = "";
+        }
+        var pos = mess.indexOf(": "a);
```

Comment:
Expected ')' and instead saw 'a'.<br>Missing semicolon.<br>Expected an identifier and instead saw ')'.<br>Expected an assignment or function call and instead saw an expression.